src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -90,12 +92,20 @@ public void run() {
     final RepairSegment segment = context.storage.getRepairSegment(segmentId).get();
     Thread.currentThread().setName(clusterName + ":" + segment.getRunId() + ":" + segmentId);
 
-    runRepair();
-    long delay = intensityBasedDelayMillis(intensity);
-    try {
-      Thread.sleep(delay);
-    } catch (InterruptedException e) {
-      LOG.warn("Slept shorter than intended delay.");
+    if (context.storage.takeLeadOnSegment(segmentId)) {
+      if(runRepair()) {
+        long delay = intensityBasedDelayMillis(intensity);
+        try {
+          Thread.sleep(delay);
+        } catch (InterruptedException e) {
+          LOG.warn("Slept shorter than intended delay.");
+        }
+      }
+    
+      context.storage.releaseLeadOnSegment(segmentId);
+    }
+    else {
+      postpone(context, segment, Optional.fromNullable(this.repairUnit));
```

Comment:
> all reaper nodes that fail to take lead on a segment will mark the segment as NOT_STARTED.

 i think this is a problem…?

> another example of the section of the code not being process scalable is that the `repair_segment` table suffers from the read-before-write anti-pattern because of the `fail_count` column.

could be a problem in non-elected-owners are updating fail_count.

otherwise it's not very clear code, because the elected leader is based off `repair_id`.
are there alternative LWT approaches, like removing `repair_id` (somehow) and adding a host+process uuid (the reaper instance's host+process) to `repair_segment` that LWTs can work off?